packages/toolkit/src/cloudWatchLogs/commands/tailLogGroup.ts

```diff
@@ -0,0 +1,354 @@
+//             filterPattern: response.filterPattern,
+//             startTime: response.timeRange.start,
+//             endTime: response.timeRange.end,
+//         }
+//     }
+
+//     return logData
+// }
+
+export async function prepareDocument(uri: vscode.Uri) {
+    try {
+        const doc = await vscode.workspace.openTextDocument(uri)
+        await vscode.window.showTextDocument(doc, { preview: false })
+        await vscode.languages.setTextDocumentLanguage(doc, 'log')
+    } catch (err) {
+        if (CancellationError.isUserCancelled(err)) {
+            throw err
+        }
+
+        throw ToolkitError.chain(
+            err,
+            localize('AWS.cwl.searchLogGroup.errorRetrievingLogs', 'Failed to get logs for {0}', uri.toString())
+        )
+    }
+}
+
+// /**
+//  * "Search Log Group" command
+//  *
+//  * @param registry
+//  * @param source Telemetry "source" name
+//  * @param logData
+//  */
+// export async function searchLogGroup(
+//     registry: LogDataRegistry,
+//     source: string,
+//     logData?: { regionName: string; groupName: string }
+// ): Promise<void> {
+//     await telemetry.cloudwatchlogs_open.run(async span => {
+//         const wizard = new SearchLogGroupWizard(logData)
+//         span.record({ cloudWatchResourceType: 'logGroup', source: source })
+//         const response = await wizard.run()
+//         if (!response) {
+//             throw new CancellationError('user')
+//         }
+
+//         const userResponse = handleWizardResponse(response, registry)
+//         const uri = createURIFromArgs(userResponse.logGroupInfo, userResponse.parameters)
+//         await prepareDocument(uri, userResponse, registry)
+//     })
+// }
+
+export async function tailLogGroup(
+    registry: LogDataRegistry,
+    source: string,
+    logData?: { regionName: string; groupName: string; groupArn: string }
+) {
+    const wizard = new TailLogGroupWizard(logData)
+    const response = await wizard.run()
+    if (!response) {
+        throw new CancellationError('user')
+    }
+    const logGroupInfo: CloudWatchLogsGroupInfo = {
+        groupName: response.submenuResponse.data,
+        regionName: response.submenuResponse.region,
+    }
+    const uri = createURIFromArgs(logGroupInfo, {})
+    await prepareDocument(uri)
+
+    const client = new CloudWatchLogsClient({ region: logGroupInfo.regionName })
+
+    const command = new StartLiveTailCommand({
+        logGroupIdentifiers: [formatGroupArn(logData?.groupArn!)],
+    })
+
+    try {
+        const response = await client.send(command)
+        handleResponseAsync(response, uri)
+    } catch (err) {
+        // Pre-stream exceptions are captured here
+        console.log(err)
+    }
+}
+
+function formatGroupArn(groupArn: string): string {
+    return groupArn.endsWith(':*') ? groupArn.substring(0, groupArn.length - 2) : groupArn
+}
+
+export async function handleResponseAsync(response: StartLiveTailCommandOutput, uri: vscode.Uri) {
```

Comment:
This is the main new code. The other code looks pretty redundant, right? 

To start with, I think the "tail" feature can just be an option/button on the existing Log Stream view. If we also add a new "Tail" command, it should use the same codepath with that option enabled. That should avoid most of the redundant code.

Reply:
Sorry, this PR was created by mistake. We were testing stuff out for a quick POC.

Reply:
Nevertheless, please consider the above comments in the non-POC case.